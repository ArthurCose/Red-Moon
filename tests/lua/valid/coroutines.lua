local co
co = coroutine.create(function(i)
  print("running:", coroutine.running())
  print("yieldable:", coroutine.isyieldable())
  print("status:", coroutine.status(co)) -- resumed

  while coroutine.yield(i) do
    i = i + 1
  end

  return "end"
end)

print("running:", coroutine.running())
print("yieldable:", coroutine.isyieldable())
print("status:", coroutine.status(co)) -- suspended
print(coroutine.resume(co, 1))         -- 1
print("status:", coroutine.status(co)) -- suspended
print(coroutine.resume(co, true))      -- 2
print(coroutine.resume(co, true))      -- 3
print(coroutine.resume(co, false))     -- end
print("status:", coroutine.status(co)) -- dead
print("running:", coroutine.running())

print("\nyield from inner call:")
co = coroutine.create(function()
  (function()
    print("a")
    coroutine.yield()
    print("b")
  end)()
  print("c")
end)
coroutine.resume(co)
print("yielded")
coroutine.resume(co)
