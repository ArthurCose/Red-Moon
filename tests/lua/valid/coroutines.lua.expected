running:	nil	true
yieldable:	false
status:	suspended
running:	thread	false
yieldable:	true
status:	running
true	1
status:	suspended
true	2
true	3
true	end
status:	dead
running:	nil	true

yield from inner call:
a
yielded
b
c
